### PR TITLE
Add preferences for RHEL 10 Beta and CentOS 10 Stream

### PR DESCRIPTION
**Pull request by 0xFelix**
**What this PR does / why we need it**:

This adds preferences intended to be used with the RHEL 10 Beta and CentOS 10 Stream.

Functional tests are not added yet because containerdisks for these OSes are not yet available.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Added preferences for RHEL 10 Beta and CentOS 10 Stream
```

### CODE REVIEW

File: preferences/centos/10_stream/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+kind: Kustomization
+
+resources:
+  - ../base
+
+components:
+  - ./metadata
+  - ./requirements
+  - ../../components/rng
+  - ../../components/disk-dedicatediothread
```
Comment (dominikholler) [inline preferences/centos/10_stream/kustomization.yaml:12]: @asamalik should the CentOS Stream 10 cloud image but with BIOS, like CentOS Stream 9, or with UEFI and secure boot like RHEL ?

Comment (0xFelix) [reply preferences/centos/10_stream/kustomization.yaml:12]: Right now it is a straight copy from CentOS Stream 9.

Comment (dominikholler) [reply preferences/centos/10_stream/kustomization.yaml:12]: @tdawson should the CentOS Stream 10 cloud image but with BIOS, like CentOS Stream 9, or with UEFI and secure boot like RHEL ?

**Comment by tdawson** [comment]
CentOS Stream 10 can  do UEFI.  That part is fine.
We just don't have a valid shim anymore, so we can't do the secure boot.
We have a shim, so you can pretend you are doing secure boot, but any real test will fail.

**Comment by 0xFelix** [comment]
@tdawson To clarify a bit, Dominik was asking about the CentOS Stream 10 GenericCloud image. In CentOS Stream 9 the [image](https://cloud.centos.org/centos/9-stream/x86_64/images/CentOS-Stream-GenericCloud-9-latest.x86_64.qcow2) supports only BIOS boot. What can we expect from the CentOS Stream 10 GenericCloud image?

**Comment by tdawson** [comment]
The generic cloud images should be built soon.  We're having a problem with the new anaconda's kickstart.
https://issues.redhat.com/browse/CS-2358

Once they are built, they will be in the compose area.  I don't have a date when they will be pushed to the official areas.

File: preferences/rhel/10/amd64/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+resources:
+  - ../../base
+
+components:
+  - ./metadata
+  - ./requirements
+  - ../../../components/disk-dedicatediothread
+  - ../../../components/rng
+  - ../../../components/secureboot
```
Comment (dominikholler) [inline preferences/rhel/10/amd64/kustomization.yaml:13]: @0xFelix can you please disable secure boot in the initial version?

Comment (0xFelix) [reply preferences/rhel/10/amd64/kustomization.yaml]: Addressed

**Comment by 0xFelix** [comment]
@dominikholler Functest for RHEL10 was added.

**Comment by dominikholler** [comment]
@0xFelix thank you, looks great!

**Review by 0xFelix — COMMENTED** [review]
/approve

**Pull request merged**